pkg/server/server.go

```diff
@@ -378,29 +378,31 @@ func (s *Server) SendVerificationLink(c *gin.Context, email string) {
 }
 
 // VerifyEmail marks a user email as verified
-func (s *Server) VerifyEmail(c *gin.Context) {
+func (s *Server) VerifyEmail(c *gin.Context, redirectURL string) {
 	jwtUser, exists := c.Get(types.JWTUserCredentialsKey)
 	if !exists {
 		s.redirectError(c, errors.ErrInvalidAccessToken)
+		c.Redirect(http.StatusUnauthorized, redirectURL)
 		return
 	}
 	jwtUserCredentials := jwtUser.(*models.UserCredentials)
 
-	updatedUserInfo, err := usermanager.UpdateUserDetails(s.userStore, jwtUserCredentials)
-	if err != nil {
-		s.redirect(c, err)
-	}
-
-	tgr := &jwtmanager.TokenGenerateRequest{
-		UserInfo:       updatedUserInfo,
-		AccessTokenExp: time.Minute * 10,
+	if jwtUserCredentials.UnverifiedEmail != nil {
+		*jwtUserCredentials.Email = jwtUserCredentials.GetUnverifiedEmail()
```

Comment:
Is there any difference between jwtUserCredentials.UnverifiedEmail and jwtUserCredentials.GetUnverifiedEmail() ?

Reply:
`UnverifiedEmail` is a pointer to string and `GetUnverifiedEmail` returns its values as string and `""` otherwise

Reply:
Then, *jwtUserCredentials.UnverifiedEmail should give the value?
Main question is do we need an accessor method?

Reply:
The accessor method will just return the zero value if we try to access it through a nil pointer. It just ensures that we do not get a nil pointer exception in worst case. 
Although we are already checking it in this case as we have to respond to UI accordingly.